File: FE-customer/src/types/models/combo.ts
import type { ProductStatus } from "../enums/product.enum";

export interface ComboItem {
  id: number;
  productId: number;
  productName: string;
  productImageUrl: string | null;
  productPrice: number;
  quantity: number;
  displayOrder: number;
}

export type Combo = {
  id: number;
  name: string;
  description: string;
  basePrice: number;
  originalPrice: number;
  imageUrls: string[];
  items: ComboItem[];
  status: ProductStatus;
  createdAt: string;
  updatedAt: string;
};

File: BE/src/main/java/com/siupo/restaurant/model/Combo.java
package com.siupo.restaurant.model;

import com.siupo.restaurant.enums.EProductStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "combos")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Combo {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(length = 2000)
    private String description;

    @Column(nullable = false)
    private Double basePrice;

    @OneToMany(mappedBy = "combo", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<ComboItem> items;

    @OneToMany(mappedBy = "combo", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<ComboImage> images;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private EProductStatus status = EProductStatus.AVAILABLE;

    @CreationTimestamp
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;
}
